packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/quanticSearchBoxInput.js

```diff
@@ -0,0 +1,344 @@
+    // Reset selection set to true for key pressed other than ARROW keys and ENTER.
+    this.sendInputValueChangeEvent(this.input.value, true);
+  }
+
+  onSubmit(event) {
+    event.stopPropagation();
+    this.sendInputValueChangeEvent(this.input.value, false);
+    this.sendSubmitSearchEvent();
+    this.input.blur();
+  }
+
+  /**
+   * Prevent default behavior of enter key, on textArea, to prevent skipping a line.
+   * @param {KeyboardEvent} event
+   */
+  onKeydown(event) {
+    if (event.key === keys.ENTER) {
+      event.preventDefault();
+    }
+  }
+
+  /**
+   * @param {KeyboardEvent} event
+   */
+  onKeyup(event) {
+    switch (event.key) {
+      case keys.ENTER:
+        this.handleEnter();
+        break;
+      case keys.ARROWUP:
+        this.suggestionList?.selectionUp();
+        break;
+      case keys.ARROWDOWN:
+        this.suggestionList?.selectionDown();
+        break;
+      default:
+        this.handleKeyValues();
+    }
+  }
+
+  onFocus() {
+    this.showSuggestions();
+    this.adjustTextAreaHeight();
+  }
+
+  onBlur() {
+    this.hideSuggestions();
+    this.collapseTextArea();
+  }
+
+  onTextAreaInput() {
+    this.handleValueChange();
+    this.adjustTextAreaHeight();
+  }
+
+  adjustTextAreaHeight() {
+    if (!this.textarea) {
+      return;
+    }
+    this.input.value = this.input.value.replace(/\n/g, '');
+    this.input.style.height = '';
+    this.input.style.whiteSpace = 'pre-wrap';
+    this.input.style.height = this.input.scrollHeight + 'px';
+  }
+
+  collapseTextArea() {
+    if (!this.textarea) {
+      return;
+    }
+    this.input.style.height = '';
+    this.input.style.whiteSpace = 'nowrap';
+  }
+
+  clearInput() {
+    this.input.value = '';
+    this.sendInputValueChangeEvent(this.input.value, false);
+    this.input.focus();
+    if (this.textarea) {
+      this.adjustTextAreaHeight();
+    }
+  }
+
+  showSuggestions() {
+    this.sendShowSuggestionsEvent();
+    this.combobox?.classList.add('slds-is-open');
+    this.combobox?.setAttribute('aria-expanded', 'true');
+  }
+
+  hideSuggestions() {
+    this.combobox?.classList.remove('slds-is-open');
+    this.combobox?.setAttribute('aria-expanded', 'false');
+    this.suggestionList?.resetSelection();
+  }
+
+  handleHighlightChange(event) {
+    this.input.value = event.detail?.rawValue;
+  }
+
+  handleSuggestionSelection(event) {
+    const textValue = event.detail;
+    this.sendSelectSuggestionEvent(textValue);
+    this.blur();
+  }
+
+  handleSuggestionListEvent = (event) => {
+    event.stopPropagation();
+    const id = event.detail;
+    this.input.setAttribute('aria-controls', id);
+  };
+
+  get searchBoxContainerClass() {
+    if (this.withoutSubmitButton) {
+      this.input?.setAttribute('aria-labelledby', 'fixed-text-label');
+      return CLASS_WITHOUT_SUBMIT;
+    }
+    this.input?.setAttribute(
+      'aria-labelledby',
+      'fixed-text-label fixed-text-addon-post'
+    );
+    return CLASS_WITH_SUBMIT;
+  }
```

Comment:
```suggestion
  get searchBoxContainerClass() {
    return `slds-combobox__form-element slds-input-has-icon ${this.withoutSubmitButton? 'slds-input-has-icon_left-right' : 'slds-input-has-icon_right slds-input-has-fixed-addon'} slds-grid`
```

Reply:
changed it to this since I need to also change the attributes:
<img width="905" alt="image" src="https://github.com/coveo/ui-kit/assets/73316533/7ca2e679-e59e-43e6-ba7c-a34303b475ec">

Reply:
The attribute should not be set in the getter.
You can easily create a new getter for this `aria-labelledby` attribute. 

for example in the JS file:

```
  get ariaLabelledByValue() {
    return `fixed-text-label ${this.withoutSubmitButton? '' : 'fixed-text-addon-post'}`
  }
```
then in the HTML you can have the following:
```
<input aria-labelledby={ariaLabelledByValue}  /> 
```

Reply:
btw what are the values given to this attribute coming from? 
these values `fixed-text-label` and `fixed-text-addon-post`.
they are supposed be Ids given to other elements.

Reply:
removed these as following investigation and tests with screen reader were deemed useless